core/templates/pages/topic-editor-page/modal-templates/rearrange-skills-in-subtopics-modal.component.ts

```diff
@@ -0,0 +1,161 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for RearrangeSkillsInSubtopicsModal.
+ */
+
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ShortSkillSummary } from 'domain/skill/short-skill-summary.model';
+import { TopicUpdateService } from 'domain/topic/topic-update.service';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { Subscription } from 'rxjs';
+import { SubtopicValidationService } from '../services/subtopic-validation.service';
+import { TopicEditorStateService } from '../services/topic-editor-state.service';
+import { Topic } from 'domain/topic/TopicObjectFactory';
+import { Subtopic } from 'domain/topic/subtopic.model';
+import { ConfirmOrCancelModal } from 'components/common-layout-directives/common-elements/confirm-or-cancel-modal.component';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { CdkDragDrop, moveItemInArray, transferArrayItem } from '@angular/cdk/drag-drop';
+
+@Component({
+  selector: 'oppia-rearrange-skills-in-subtopics-modal',
+  templateUrl: './rearrange-skills-in-subtopics-modal.component.html'
+})
+export class RearrangeSkillsInSubtopicsModalComponent
+ extends ConfirmOrCancelModal implements OnInit, OnDestroy {
+  topic: Topic;
+  subtopics: Subtopic[];
+  uncategorizedSkillSummaries: ShortSkillSummary[];
+  skillSummaryToMove: ShortSkillSummary;
+  oldSubtopicId: number | null = null;
+  errorMsg: string;
+  editableName: string;
+  selectedSubtopicId: number;
+
+  SKILL_EDITOR_URL_TEMPLATE = '/skill_editor/<skillId>';
+  directiveSubscriptions = new Subscription();
+
+  constructor(
+    private subtopicValidationService: SubtopicValidationService,
+    private topicEditorStateService: TopicEditorStateService,
+    private topicUpdateService: TopicUpdateService,
+    private urlInterpolationService: UrlInterpolationService,
+    private ngbActiveModal: NgbActiveModal,
+  ) {
+    super(ngbActiveModal);
+  }
+
+  initEditor(): void {
+    this.topic = this.topicEditorStateService.getTopic();
+    this.subtopics = this.topic.getSubtopics();
+    this.uncategorizedSkillSummaries = (
+      this.topic.getUncategorizedSkillSummaries());
+  }
+
+  getSkillEditorUrl(skillId: string): string {
+    return this.urlInterpolationService.interpolateUrl(
+      this.SKILL_EDITOR_URL_TEMPLATE, {
+        skillId: skillId
+      }
+    );
+  }
+
+  /**
+   * @param {string|null} oldSubtopicId - The id of the subtopic from
+   *    which the skill is to be moved, or null if the origin is the
+   *    uncategorized section.
+   * @param {SkillSummary} skillSummary - The summary of the skill that
+   *    is to be moved.
+   */
+  onMoveSkillStart(
+      oldSubtopicId: number | null, skillSummary: ShortSkillSummary): void {
+    this.skillSummaryToMove = skillSummary;
+    this.oldSubtopicId = oldSubtopicId ? oldSubtopicId : null;
+  }
```

Comment:
The parameter description of `oldSubtopicId` says that it can either be string or null, however, in the function the defined type of param `oldSubtopicId` is number or null! Please check!